components/supervisor/pkg/supervisor/supervisor.go

```diff
@@ -1669,3 +1670,25 @@ func handleExit(ec *int) {
 	log.WithField("exitCode", exitCode).Debug("supervisor exit")
 	os.Exit(exitCode)
 }
+
+func waitForIde(parent context.Context, ideReady *ideReadyState, desktopIdeReady *ideReadyState, timeout time.Duration) {
```

Comment:
@mustard-mh Should we clean it up to remove duplication?

Maybe introduce intermediate struct which we use here and pass to a task manager? It would have `Wait` function with a doc for us for the future?